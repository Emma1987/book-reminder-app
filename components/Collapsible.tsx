import { PropsWithChildren, useEffect, useRef, useState } from 'react';
import { Animated, Easing, StyleSheet, TouchableOpacity, View, useAnimatedValue } from 'react-native';
import Ionicons from '@expo/vector-icons/Ionicons';

import { ThemedText } from '@/components/ThemedText';
import { Colors } from '@/constants/Colors';

type CollapsibleProps = PropsWithChildren<{
    title: string;
    isCollapsibleOpen: boolean;
}>;

export function Collapsible({ children, title, isCollapsibleOpen = false }: CollapsibleProps) {
    const [isOpen, setIsOpen] = useState(isCollapsibleOpen);
    const animatedHeight = useAnimatedValue(isCollapsibleOpen ? 1 : 0);
    const [contentHeight, setContentHeight] = useState(0);

    useEffect(() => {
        Animated.timing(animatedHeight, {
            toValue: isOpen ? 1 : 0,
            duration: 300,
            easing: Easing.linear,
            useNativeDriver: false,
        }).start();
    }, [isOpen, contentHeight]);

    useEffect(() => {
        setIsOpen(isCollapsibleOpen);
    }, [isCollapsibleOpen]);

    const toggle = () => setIsOpen(!isOpen);

    return (
        <View>
            <TouchableOpacity
                style={styles.heading}
                onPress={toggle}
                activeOpacity={0.8}
                accessibilityRole="button"
                accessibilityState={{ expanded: isOpen }}
                accessibilityLabel={`${title} section`}
            >
                <Ionicons
                    name={isOpen ? 'chevron-down' : 'chevron-forward-outline'}
                    size={18}
                    color={Colors.icon}
                />
                <ThemedText type="defaultSemiBold">{title}</ThemedText>
            </TouchableOpacity>
            <Animated.View
                style={[
                    styles.content,
                    {
                        height: animatedHeight.interpolate({
                            inputRange: [0, 1],
                            outputRange: [0, contentHeight || 0],
                        }),
                    },
                ]}
            >
                <View
                    style={styles.animatedView}
                    onLayout={(event) => {
                        const measuredHeight = event.nativeEvent.layout.height;
                        setContentHeight(measuredHeight);
                    }}
                >
                    {children}
                </View>
            </Animated.View>
        </View>
    );
}

const styles = StyleSheet.create({
    heading: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 6,
    },
    content: {
        marginTop: 6,
        overflow: 'hidden',
    },
    animatedView: {
        minHeight: 1
    },
});
