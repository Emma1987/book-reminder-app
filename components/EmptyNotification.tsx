import React from 'react';
import { View, Image, StyleSheet } from 'react-native';

import { ThemedText } from '@/components/ThemedText';
import i18n from '@/i18n/translations';

export function EmptyNotification() {
    return (
        <View style={styles.container}>
            <Image
                source={require('@/assets/images/no-notification.png')}
                style={styles.image}
                resizeMode="contain"
                accessibilityRole="image"
                accessibilityLabel={i18n.t('notifications_screen.empty_notifications')}
            />
            <ThemedText type="default" textAlign="center">
                {i18n.t('notifications_screen.empty_notifications')}
            </ThemedText>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 50,
        alignItems: 'center',
    },
    image: {
        height: 200,
    },
});
