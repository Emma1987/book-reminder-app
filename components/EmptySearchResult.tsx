import React from 'react';
import { View, Image, StyleSheet } from 'react-native';

import { ThemedText } from '@/components/ThemedText';
import i18n from '@/i18n/translations';

export function EmptySearchResult() {
    return (
        <View style={styles.container}>
            <Image
                source={require('@/assets/images/no-result.png')}
                style={styles.image}
                resizeMode="contain"
                accessibilityRole="image"
                accessibilityLabel={i18n.t('empty_search_result.no_result_icon')}
            />
            <ThemedText type="default" textAlign="center">
                {i18n.t('empty_search_result.title')}
            </ThemedText>
            <ThemedText type="info" textAlign="center" style={styles.descriptionInfo}>
                {i18n.t('empty_search_result.help')}
            </ThemedText>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 50,
        alignItems: 'center',
    },
    image: {
        height: 200,
    },
    descriptionInfo: {
        width: '70%',
        fontStyle: 'italic',
    },
});
