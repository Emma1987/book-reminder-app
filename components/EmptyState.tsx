import React from 'react';
import { View, Image, StyleSheet, Text, TouchableOpacity } from 'react-native';
import { ThemedText } from '@/components/ThemedText';
import { Colors } from '@/constants/Colors';
import i18n from '@/i18n/translations';

type EmptyStateProps = {
    title: string;
    description: string;
    imageSource: any;
    buttonText?: string;
    onButtonPress?: () => void;
};

export function EmptyState({ title, description, imageSource, buttonText, onButtonPress }: EmptyStateProps) {
    return (
        <View style={styles.container}>
            <Image
                source={imageSource}
                style={styles.image}
                resizeMode="contain"
                accessibilityRole="image"
                accessibilityLabel={i18n.t('empty_state.illustration_label')}
            />
            <View style={styles.description}>
                <ThemedText type="defaultSemiBold" textAlign="center">
                    {title}
                </ThemedText>
                <ThemedText type="info" textAlign="center" style={styles.descriptionInfo}>
                    {description}
                </ThemedText>
                {buttonText && onButtonPress && (
                    <TouchableOpacity
                        style={styles.button}
                        onPress={onButtonPress}
                        accessibilityRole="button"
                        accessibilityLabel={buttonText}
                    >
                        <Text>{buttonText}</Text>
                    </TouchableOpacity>
                )}
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 20,
        alignItems: 'center',
    },
    image: {
        height: 200,
    },
    description: {
        width: '70%',
    },
    descriptionInfo: {
        fontStyle: 'italic',
    },
    button: {
        alignSelf: 'center',
        marginTop: 20,
        paddingVertical: 10,
        paddingHorizontal: 20,
        backgroundColor: Colors.secondaryColor,
        borderRadius: 5,
    },
});
