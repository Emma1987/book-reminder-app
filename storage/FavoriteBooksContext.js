import React, { createContext, useState, useEffect } from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';

export const FavoriteBooksContext = createContext();

export const FavoriteBooksProvider = ({ children }) => {
    const [favorites, setFavorites] = useState([]);

    useEffect(() => {
        loadFavorites();
    }, []);

    const loadFavorites = async () => {
        const storedFavorites = await AsyncStorage.getItem('favorites');
        if (storedFavorites) setFavorites(JSON.parse(storedFavorites));
    };

    const addFavorite = async (book) => {
        const updatedFavorites = [...favorites, book];
        setFavorites(updatedFavorites);
        await AsyncStorage.setItem('favorites', JSON.stringify(updatedFavorites));
    };

    const removeFavorite = async (bookId) => {
        const updatedFavorites = favorites.filter((book) => book.id !== bookId);
        setFavorites(updatedFavorites);
        await AsyncStorage.setItem('favorites', JSON.stringify(updatedFavorites));
    };

    const isFavorite = (bookId) => favorites.some((book) => book.id === bookId);

    return (
        <FavoriteBooksContext.Provider value={{ favorites, addFavorite, removeFavorite, isFavorite }}>
            {children}
        </FavoriteBooksContext.Provider>
    );
};
