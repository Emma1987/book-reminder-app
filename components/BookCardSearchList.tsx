import React, { useContext } from 'react';
import { Alert, Image, Keyboard, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

import { ThemedText } from '@/components/ThemedText';
import { BookType } from '@/components/types';
import { Colors } from '@/constants/Colors';
import { isPublished } from '@/helpers/BookHelper';
import { formatDateStr } from '@/helpers/DateHelper';
import { scheduleBookReleaseNotification } from '@/helpers/BookNotificationHelper';
import { FavoriteBooksContext } from '@/storage/FavoriteBooksContext';
import { NotificationContext } from '@/storage/NotificationContext';

type BookCardSearchListProps = {
    book: BookType;
    onSeeDetails: (book: BookType) => void;
};

export function BookCardSearchList({ book, onSeeDetails }: BookCardSearchListProps) {
    const { addFavorite, removeFavorite, isFavorite } = useContext(FavoriteBooksContext);
    const { addNotification } = useContext(NotificationContext);

    const getNotified = () => {
        const alertContent = `ðŸ“š The book "${book.title}" is set to be released on ${formatDateStr(book.releaseDateRaw, 'monthFirst')}.\n\nWould you like to receive a notification on the release date?`;

        Alert.alert('Stay Updated!', alertContent, [
            {
                text: 'No, thanks',
                style: 'cancel',
                onPress: () => console.log('Notification declined'),
            },
            {
                text: 'Yes, notify me!',
                style: 'default',
                onPress: async () => {
                    scheduleBookReleaseNotification(book, addNotification);
                },
            },
        ]);
    };

    const toggleFavorite = () => {
        Keyboard.dismiss();

        if (isFavorite(book.id)) {
            removeFavorite(book.id);
        } else {
            if (!isPublished(book) && book.releaseDateRaw !== null) {
                getNotified();
            }

            addFavorite(book);
        }
    };

    return (
        <View style={styles.container}>
            {/* Book Cover */}
            <Image
                source={
                    book.coverImage ? { uri: book.coverImage } : require('@/assets/images/book-cover-placeholder.jpg')
                }
                style={styles.bookCover}
                accessibilityRole="image"
                accessibilityLabel="Book cover"
            />

            {/* Book Description */}
            <View style={styles.bookDescription}>
                <ThemedText type="subtitle">{book.title}</ThemedText>
                <ThemedText type="defaultSemiBold">{book.authors?.[0] ?? ''}</ThemedText>
                <Text>Release date: {formatDateStr(book.releaseDateRaw)}</Text>
            </View>

            {/* Action Icons */}
            <View style={styles.actions}>
                <TouchableOpacity
                    onPress={toggleFavorite}
                    accessibilityLabel="Add to favorites"
                    accessibilityHint="Adds this book to your favorites list"
                >
                    {isFavorite(book.id) && <Ionicons name="heart" size={25} color={Colors.red} />}
                    {!isFavorite(book.id) && <Ionicons name="heart-outline" size={25} color={Colors.gray} />}
                </TouchableOpacity>
                <TouchableOpacity
                    onPress={() => onSeeDetails(book)}
                    accessibilityLabel="View book details"
                    accessibilityHint={`View more details about ${book.title}`}
                >
                    <Ionicons name="information-circle-outline" size={25} color={Colors.gray} />
                </TouchableOpacity>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        padding: 10,
        borderBottomWidth: 1,
        borderColor: Colors.white,
    },
    bookCover: {
        width: 60,
        height: 90,
    },
    bookDescription: {
        flex: 1,
        padding: 10,
    },
    actions: {
        width: 60,
        justifyContent: 'space-between',
        alignItems: 'center',
    },
});
