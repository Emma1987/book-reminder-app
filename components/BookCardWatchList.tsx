import React, { useContext } from 'react';
import { Alert, Image, StyleSheet, TouchableOpacity, View } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

import { ThemedText } from '@/components/ThemedText';
import { BookType } from '@/components/types';
import { Colors } from '@/constants/Colors';
import { FavoriteBooksContext } from '@/storage/FavoriteBooksContext';

type BookCardWatchListProps = {
    book: BookType;
};

export function BookCardWatchList({ book }: BookCardWatchListProps) {
    const { removeFavorite } = useContext(FavoriteBooksContext);

    const handleRemoveFavorite = () => {
        Alert.alert('Remove Favorite', `Are you sure you want to remove "${book.title}" from your favorites?`, [
            {
                text: 'Cancel',
                style: 'cancel',
                onPress: () => {},
            },
            {
                text: 'Remove',
                style: 'destructive',
                onPress: () => removeFavorite(book.id),
            },
        ]);
    };

    return (
        <View style={styles.container}>
            {/* Book Cover */}
            <Image
                source={
                    book.coverImage ? { uri: book.coverImage } : require('@/assets/images/book-cover-placeholder.jpg')
                }
                style={styles.bookCover}
                accessibilityRole="image"
                accessibilityLabel="Book cover"
            />

            {/* Book Description */}
            <View style={styles.bookDescription}>
                <ThemedText type="defaultSemiBold">{book.title}</ThemedText>
                <ThemedText type="default">{book.authors?.[0] ?? ''}</ThemedText>
                <ThemedText type="info">Publication: {book.publicationDateStr}</ThemedText>
            </View>

            {/* Action Icons */}
            <View style={styles.actions}>
                <TouchableOpacity
                    onPress={handleRemoveFavorite}
                    accessibilityLabel="Remove from favorites"
                    accessibilityHint="Removes this book from your favorites list"
                >
                    <Ionicons name="trash-outline" size={20} color={Colors.red} />
                </TouchableOpacity>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        padding: 10,
        backgroundColor: Colors.whitesmoke,
        borderRadius: 5,
        marginBottom: 5,
    },
    bookCover: {
        width: 40,
        height: 60,
    },
    bookDescription: {
        flex: 1,
        paddingLeft: 10,
    },
    actions: {
        width: 30,
        alignItems: 'center',
    },
});
