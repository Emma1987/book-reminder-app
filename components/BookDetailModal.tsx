import React, { useMemo, useContext, useRef } from 'react';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { GestureHandlerRootView } from 'react-native-gesture-handler';
import BottomSheet, { BottomSheetScrollView } from '@gorhom/bottom-sheet';
import { Ionicons } from '@expo/vector-icons';

import { Separator } from '@/components/Separator';
import { ThemedText } from '@/components/ThemedText';
import BookType from '@/components/types';
import { Colors } from '@/constants/Colors';
import { FavoriteBooksContext } from '@/storage/FavoriteBooksContext';

export function BookDetailModal({ visible, onClose, book }) {
    const { addFavorite, removeFavorite, isFavorite } = useContext(FavoriteBooksContext);
    const bottomSheetRef = useRef<BottomSheet>(null);
    const snapPoints = useMemo(() => ['50%', '90%'], []);

    const closeModal = () => {
        onClose();
        bottomSheetRef.current.close();
    }

    const toggleFavorite = () => {
        if (isFavorite(book.id)) {
            removeFavorite(book.id);
        } else {
            addFavorite(book);
        }
    };

    return (
        <BottomSheet
            ref={bottomSheetRef}
            index={visible ? 0 : -1}
            snapPoints={snapPoints}
            onClose={closeModal}
            enablePanDownToClose={true}
            enableDynamicSizing={false}
        >
            <BottomSheetScrollView style={styles.bottomSheetView}>
                {/* Book Cover */}
                {book.coverImage &&
                    <Image source={{ uri: book.coverImage }} style={styles.bookCover} />
                }

                {/* Title, author and publication date */}
                <ThemedText type="subtitle-center">{book.title}</ThemedText>
                <ThemedText type="defaultSemiBold-center" style={styles.mb10}>Author: {book.authors?.join(', ')}</ThemedText>
                <ThemedText type="info-center" style={styles.mb10}>Publication date: {book.publicationDateStr}</ThemedText>

                {/* Language */}
                {book.language && 
                    <ThemedText type="info-center" style={styles.mb10}>Language: {book.language.toUpperCase()}</ThemedText>
                }

                {/* ISBN */}
                {book.isbn10 && <ThemedText type="info-center">ISBN10: {book.isbn10}</ThemedText>}
                {book.isbn13 && <ThemedText type="info-center">ISBN13: {book.isbn13}</ThemedText>}

                {/* Description */}
                {book.description &&
                    <View>
                        <Separator />
                        <Text style={styles.description}>{book.description}</Text>
                    </View>
                }

                {/* Add to favs Button */}
                {isFavorite(book.id) && 
                    <TouchableOpacity style={[styles.button, styles.removeFavButton]} onPress={toggleFavorite}>
                        <Text style={styles.buttonText}>Remove from favorites</Text>
                    </TouchableOpacity>
                }
                {!isFavorite(book.id) && 
                    <TouchableOpacity style={[styles.button, styles.addFavButton]} onPress={toggleFavorite}>
                        <Text style={styles.buttonText}>Add to favorites</Text>
                    </TouchableOpacity>
                }
            </BottomSheetScrollView>
        </BottomSheet>
    );
}

const styles = StyleSheet.create({
    bottomSheetView: {
        padding: 20,
    },
    bookCover: {
        width: 100,
        height: 150,
        alignSelf: 'center',
        marginBottom: 16,
    },
    textCenter: {
        textAlign: 'center'
    },
    mb10: {
        marginBottom: 10,
    },
    description: {
        fontSize: 14,
        textAlign: 'justify',
        marginBottom: 50,
    },
    button: {
        paddingVertical: 10,
        paddingHorizontal: 20,
        marginBottom: 50,
        borderRadius: 5,
        width: 150,
        alignSelf: 'center',
    },
    buttonText: {
        color: Colors.white,
        fontWeight: 'bold',
        textAlign: 'center',
    },
    addFavButton: {
        backgroundColor: Colors.red,
    },
    removeFavButton: {
        backgroundColor: Colors.gray,
    },
});
