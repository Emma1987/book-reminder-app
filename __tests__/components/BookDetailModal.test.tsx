// BookDetailModal.test.tsx
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { itBook } from '@/__tests__/__fixtures__/fixtures';
import { BookDetailModal } from '@/components/BookDetailModal';
import { FavoriteBooksContext } from '@/storage/FavoriteBooksContext';

describe('BookDetailModal', () => {
    it('renders book details correctly', () => {
        const { getByText, debug } = render(
            <FavoriteBooksContext.Provider value={{addFavorite: jest.fn(), removeFavorite: jest.fn(), isFavorite: jest.fn()}}>
                <BookDetailModal visible={true} onClose={jest.fn()} book={itBook} />
            </FavoriteBooksContext.Provider>
        );

        expect(getByText('It')).toBeTruthy();
        expect(getByText('Author: Stephen King')).toBeTruthy();
        expect(getByText('Release date: 1 Oct 1987')).toBeTruthy();
        expect(getByText('ISBN10: 0450411435')).toBeTruthy();
        expect(getByText('ISBN13: 9780450411434')).toBeTruthy();
        expect(getByText('Welcome to Derry, Maine...')).toBeTruthy();
    });

    it('calls addFavorite when Add to favorites button is pressed', () => {
        const mockAddFavorite = jest.fn();
        
        const { getByText } = render(
            <FavoriteBooksContext.Provider value={{addFavorite: mockAddFavorite, removeFavorite: jest.fn(), isFavorite: jest.fn()}}>
                <BookDetailModal visible={true} onClose={jest.fn()} book={itBook} />
            </FavoriteBooksContext.Provider>
        );

        fireEvent.press(getByText('Add to favorites'));
        expect(mockAddFavorite).toHaveBeenCalledWith(itBook);
    });

    it('calls removeFavorite when Remove from favorites button is pressed', () => {
        const mockRemoveFavorite = jest.fn();
        const mockIsFavorite = jest.fn();
        mockIsFavorite.mockReturnValue(true);

        const { getByText } = render(
            <FavoriteBooksContext.Provider value={{addFavorite: jest.fn(), removeFavorite: mockRemoveFavorite, isFavorite: mockIsFavorite}}>
                <BookDetailModal visible={true} onClose={jest.fn()} book={itBook} />
            </FavoriteBooksContext.Provider>
        );

        fireEvent.press(getByText('Remove from favorites'));
        expect(mockRemoveFavorite).toHaveBeenCalledWith(itBook.id);
    });

    it('matches the snapshot', () => {
        const tree = render(
            <FavoriteBooksContext.Provider value={{addFavorite: jest.fn(), removeFavorite: jest.fn(), isFavorite: jest.fn()}}>
                <BookDetailModal visible={true} onClose={jest.fn()} book={itBook} />
            </FavoriteBooksContext.Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
