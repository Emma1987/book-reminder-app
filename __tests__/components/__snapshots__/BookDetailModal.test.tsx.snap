// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetailModal matches the snapshot 1`] = `
<View
  enableDynamicSizing={false}
  enablePanDownToClose={true}
  index={0}
  onClose={[Function]}
  snapPoints={
    [
      "50%",
      "90%",
    ]
  }
>
  <RCTScrollView
    style={
      {
        "padding": 20,
      }
    }
  >
    <View>
      <Text
        style={
          [
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 20,
              "fontWeight": "bold",
            },
            undefined,
            undefined,
            [
              {
                "textAlign": "center",
              },
            ],
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        It
      </Text>
      <Text
        style={
          [
            undefined,
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            [
              {
                "textAlign": "center",
              },
            ],
            [
              {
                "marginBottom": 10,
              },
            ],
            undefined,
            undefined,
          ]
        }
      >
        Author: 
        Stephen King
      </Text>
      <Text
        style={
          [
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "color": "gray",
              "fontSize": 14,
            },
            [
              {
                "textAlign": "center",
              },
            ],
            [
              {
                "marginBottom": 10,
              },
            ],
            undefined,
            undefined,
          ]
        }
      >
        Publication date: 
        1987-10-01
      </Text>
      <Text
        style={
          [
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "color": "gray",
              "fontSize": 14,
            },
            [
              {
                "textAlign": "center",
              },
            ],
            [
              {
                "marginBottom": 10,
              },
            ],
            undefined,
            undefined,
          ]
        }
      >
        Language: 
        EN
      </Text>
      <Text
        style={
          [
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "color": "gray",
              "fontSize": 14,
            },
            [
              {
                "textAlign": "center",
              },
            ],
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        ISBN10: 
        0450411435
      </Text>
      <Text
        style={
          [
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "color": "gray",
              "fontSize": 14,
            },
            [
              {
                "textAlign": "center",
              },
            ],
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        ISBN13: 
        9780450411434
      </Text>
      <View>
        <View
          style={
            {
              "backgroundColor": "#ccc",
              "height": 1,
              "marginVertical": 20,
              "width": "100%",
            }
          }
        />
        <Text
          style={
            {
              "fontSize": 14,
              "marginBottom": 50,
              "textAlign": "justify",
            }
          }
        >
          Welcome to Derry, Maine...
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "center",
            "backgroundColor": "#ff645c",
            "borderRadius": 5,
            "marginBottom": 50,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 150,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Add to favorites
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
